packages/block-editor/src/components/block-edit/index.js

```diff
@@ -34,12 +37,42 @@ export default function BlockEdit( props ) {
 		'__experimentalLayout',
 		false
 	);
+	const shouldDisplayControls = useSelect(
```

Comment:
This is good but if I'm understanding properly the current "controls", we have two cases:

 - Some controls (not blocks) that should check for `__experimentalExposeControlsToChildren` (when `__experimentalShareWithChildBlocks` is passed as a prop to BlockControls)
 - Some other controls that ignore this and only work for selected (and first multi-selected blocks).

So what this mean is that a single boolean in context is not enough, we probably need two of them.

Reply:
I'm not quite following.

Just checking `__experimentalExposeControlsToChildren` should be enough here? If the parent shares controls with the child blocks, then render all controls no matter what it is. BlockControls will then decide to render based on `__experimentalShareWithChildBlocks`.

Reply:
Testing the buttons block for example: the alignment controls are rendered if a single button is selected. And selecting two buttons still displays the alignment controls.